front/src/components/boxs/chart/Chart.jsx

```diff
@@ -248,7 +249,17 @@ class Chartbox extends Component {
       } else {
         series = data.map((oneFeature, index) => {
           const oneUnit = this.props.box.units ? this.props.box.units[index] : this.props.box.unit;
-          const oneUnitTranslated = oneUnit ? this.props.intl.dictionary.deviceFeatureUnitShort[oneUnit] : null;
+          // Convert distance units if necessary
+          let displayUnit = oneUnit;
```

Comment:
Could we do this instead?

```suggestion
          const userUnit = this.props.user.distance_unit_preference;
          // Convert unit display format (using 0 as a dummy value since we only need the unit)
          const conversion = convertUnitDistance(0, oneUnit, userUnit);
          const displayUnit = conversion.unit;

```

It is much simpler as convertUnitDistance makes all the necessary checks before converting.

_Seen with @Terdious_

Reply:
Can be repeated below with other uses of convertUnitDistance